lib/IRGen/IRGenModule.cpp

```diff
@@ -127,11 +127,16 @@ IRGenModule::IRGenModule(IRGenerator &irgen,
       ClangCodeGen(createClangCodeGenerator(Context, LLVMContext, irgen.Opts,
                                             ModuleName)),
       Module(*ClangCodeGen->GetModule()), LLVMContext(Module.getContext()),
-      DataLayout(target->createDataLayout()), Triple(Context.LangOpts.Target),
+      DataLayout(target->createDataLayout()),
+      Triple(static_cast<ClangImporter *>(
+                 &*irgen.SIL.getASTContext().getClangModuleLoader())
+                 ->getTargetInfo()
+                 .getTargetOpts()
+                 .Triple),
```

Comment:
This is pretty terrible. Can we just pass it in instead? Or wrap it up in IRGenerator somewhere?